### PR TITLE
vtk-m: Major refactor of the recipe

### PR DESCRIPTION
**Pull request by vicentebolea**
- It moves conflicts from configure time to concretization time
- It leverages in new Spack CMakePackage methods to specify CMake options in a way that significantly reduce the LoC.

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]
Much cleaner. Waiting for reviewer approval before merging.

File: var/spack/repos/builtin/packages/vtk-m/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/vtk-m/package.py]: Most of these (all maybe?) can be incorporated into the corresponding variants as a when clause.

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
Can you solve the conflict?

**Comment by vicentebolea** [comment]
> Can you solve the conflict?

Done

**Comment by vicentebolea** [comment]
@alalazo ci passed :)

**Review by alalazo — APPROVED** [review]

**Pull request merged**
